### PR TITLE
Allow kebab case resource type

### PR DESCRIPTION
**Pull request by taomoh**
Before this change, the following scenario setup (note the hyphen in the resource type string `time-entries`) wouldn't be matched in `ParameterTypes#payload()` method, resulting in the scenario failing.
`When Trevor creates the valid time-entries in the timecard service`
This is due to the regex used in `ParameterType` annotation of the above method supporting words with alphanumeric characters only

### CODE REVIEW

**Review by JWeston-HO — APPROVED** [review]

**Review by stephenkaye-ho — CHANGES_REQUESTED** [review]

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/parametertypes/ParameterTypes.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public ParameterTypes(@NonNull PersonaManager personaManager,
    * @param resourceType The type of resource represented
    * @return PayloadKey
    */
-  @ParameterType("(?:the )?(\\w+) (\\w+)")
+  @ParameterType("(?:the )?(\\w+) ([\\w-]*)")
```
[the remark this answers is not in the document]
Comment (taomoh) [reply cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/parametertypes/ParameterTypes.java]: @stephenkaye-ho , changes made as discussed

File: cucumber-jparest/src/test/java/uk/gov/homeoffice/digital/sas/cucumberjparesttestapi/stepdefinitions/PayloadManagerStep.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparesttestapi.stepdefinitions;
```
Comment (stephenkaye-ho) [inline cucumber-jparest/src/test/java/uk/gov/homeoffice/digital/sas/cucumberjparesttestapi/stepdefinitions/PayloadManagerStep.java:1]: Just wondering about the file location a little. The cucumberjparesttestapi folder was to isolate the code used for running the api whereas this code is related to the tests. What do you think?

Comment (taomoh) [reply cucumber-jparest/src/test/java/uk/gov/homeoffice/digital/sas/cucumberjparesttestapi/stepdefinitions/PayloadManagerStep.java]: Good point @stephenkaye-ho . I've re-arranged the folders

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/stepdefinitions/ApiSteps.java [hunk 3]
```diff
@@ -96,7 +95,7 @@ public void personaRetrievesResourcesFromTheService(Persona persona, String reso
    */
   @When("{persona} GETs {string}{service}")
   public void personaGetsUrlFromService(Persona persona, String path, String service) {
-    JpaRestApiResponse apiResponse = this.jpaRestApiClient.get(persona, service, path);
+    JpaRestApiResponse apiResponse = this.jpaRestApiClient.get(persona, null, service, path);
```
Comment (stephenkaye-ho) [inline cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/stepdefinitions/ApiSteps.java:98]: I'm not sure why we added this param to always pass null and then grab it from the system property.

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/config/JpaTestContext.java [hunk 4]
```diff
@@ -117,4 +118,8 @@ public Interpolation interpolation(ConfigurableBeanFactory beanFactory) {
     return new Interpolation(beanFactory);
   }
 
+  @Bean
+  public Map<String, String> sharedVariables() {
```
Comment (stephenkaye-ho) [inline cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/config/JpaTestContext.java:122]: I can't see where this is used

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/api/JpaRestApiClient.java [hunk 5]
```diff
@@ -78,32 +75,25 @@ private static URI getUri(@NonNull RequestSpecification requestSpecification) {
...
-  public JpaRestApiResourceResponse create(Persona persona, String service, String resource,
-      String payload) {
+  public JpaRestApiResourceResponse create(Persona persona, String tenantId, String service,
+      String resource, String payload) {
     URI uri = getResourceUri(service, resource);
     RequestSpecification spec = given()
         .baseUri(uri.toString())
         .body(payload)
-        .queryParam(TENANT_ID_PARAM_NAME, tenantId);
+        .queryParam(TENANT_ID_PARAM_NAME,
```
Comment (stephenkaye-ho) [inline cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/api/JpaRestApiClient.java:95]: I was going to say that we should just annotate the param to not allow nulls but then I see by default it looks like we've added the param and always pass null in all of the calling methods.
It doesn't feel like a good pattern.

**Review by stephenkaye-ho — CHANGES_REQUESTED** [review]
It's in a couple of my comments. I can't shake that adding a param to the method, adding code expecting a null value, and then only ever calling the method with a new value is not the correct pattern.
I wondered if this is to allow the value to be overridden in the future. It has dawned on me after looking at the PR that we talked about how ultimately the tenant will probably come from the persona. As we already pass the persona in each of the methods of the client (get, update, create etc) then we could add the property there and add the call to get the system property. That way we only have to ensure the system property has been set before we create the first persona. This should allow us to set it in the before all. 

Let me know if you think it will be useful to discuss this further

**Comment by taomoh** [comment]
> It's in a couple of my comments. I can't shake that adding a param to the method, adding code expecting a null value, and then only ever calling the method with a new value is not the correct pattern. I wondered if this is to allow the value to be overridden in the future. It has dawned on me after looking at the PR that we talked about how ultimately the tenant will probably come from the persona. As we already pass the persona in each of the methods of the client (get, update, create etc) then we could add the property there and add the call to get the system property. That way we only have to ensure the system property has been set before we create the first persona. This should allow us to set it in the before all.
> 
> Let me know if you think it will be useful to discuss this further

tenantId method argument removed as discussed yesterday

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/config/JpaTestContext.java (hunk 4 above)
Comment (taomoh) [reply cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/config/JpaTestContext.java]: removed

File: cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/stepdefinitions/ApiSteps.java (hunk 3 above)
Comment (taomoh) [reply cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/scenarios/stepdefinitions/ApiSteps.java]: tenantId method argument removed as discussed yesterday

**Review by stephenkaye-ho — APPROVED** [review]

**Review by JWeston-HO — APPROVED** [review]

**Review by stephenkaye-ho — APPROVED** [review]

**Pull request merged**